home/views.py

```diff
@@ -290,31 +293,62 @@ def password_gen(request):
+            form = RegistrationForm(request.POST)
+            
+            if form.is_valid():
+                try:
+                    user = form.save(commit=False)  # Save user instance without committing
+                    user.set_password(form.cleaned_data['password1'])  # Hash the password
+                    user.save()  # Save the user
+                    print("User saved successfully.")
+
+                    # Generate OTP and send email
+                    otp = random.randint(100000, 999999)
+                    email = form.cleaned_data.get('email')
+                    send_mail(
+                        subject="User Data",
+                        message=(
+                            f"Hello from HardHat Enterprise! Verify Your Mail with the OTP: \n{otp}\n"
+                            "If you didn't request an OTP or open an account with us, please contact us at your earliest convenience.\n\n"
+                            "Regards, \nHardhat Enterprises"
+                        ),
+                        from_email="deakinhardhatwebsite@gmail.com",
+                        recipient_list=[email],
+                        fail_silently=False,
+                    )
+                    print(f"OTP sent to {email}.")
+
+                    # Redirect to verify token page with context
+                    messages.success(request, "Account created successfully! Check your email for the OTP.")
+                    return render(
+                        request,
+                        'accounts/verify_token.html',
+                        {'otp': otp, 'first_name': user.first_name, 'last_name': user.last_name, 'email': user.email}
+                    )
+                except Exception as e:
+                    print(f"Error saving user or sending email: {e}")
+                    print(traceback.format_exc())  # Print detailed traceback
+                    messages.error(request, "An error occurred while creating the account. Please try again.")
+            else:
+                print("Form is invalid. Errors:")
+                print(form.errors)  # Debugging log for form errors
+                messages.error(request, "Please fix the errors below.")
         else:
-            print("Registration failed!")
-    else:
-        form = RegistrationForm()
- 
-    context = { 'form': form }
-    return render(request, 'pages/index.html', context)
+            print("GET request received for registration.")
+            form = RegistrationForm()
+
+        context = {'form': form}
+        return render(request, 'accounts/sign-up.html', context)
+
+    except Exception as e:
+        # Catch any unexpected errors and print to the terminal
+        print(f"Unexpected error in register view: {e}")
+        print(traceback.format_exc())
+        messages.error(request, "An unexpected error occurred. Please try again later.")
+        return render(request, 'accounts/sign-up.html', {'form': RegistrationForm()})
```

Comment:
currently the register view is unused and mostly a duplicate of the signup view, could you merge them into 1 view

Reply:
Have merged them now